### PR TITLE
Improve browser stealer & add SQLite lib detection

### PR DESCRIPTION
**Pull request by Still34**
## Summary

This PR adds detection for SQL statements related to cookie accesses, fixes one of the regexes, as well as adding rudimentary support for detection static sqlite3/cppsqlite3 lib linking. The relevant samples will be submitted to the test repo once I get the go ahead from the maintainers.

### CODE REVIEW

**Review by williballenthin — APPROVED** [review]
logic looks great, excited for the referenced file so we can merge this

thank you!

File: linking/static/sqlite3/linked-against-cppsqlite.yml
Comment (mr-tz) [inline linking/static/sqlite3/linked-against-cppsqlite.yml:1]: Rename rule file to match the rule name, expected: "linked-against-cppsqlite3.yml", found: "linked-against-cppsqlite.yml"

File: collection/browser/gather-chrome-based-browser-login-information.yml [hunk 1]
```diff
@@ -4,18 +4,21 @@ rule:
     namespace: collection/browser
     authors:
       - "@_re_fox"
+      - still@teamt5.org
     scope: function
     att&ck:
       - Credential Access::Credentials from Password Stores::Credentials from Web Browsers [T1555.003]
     examples:
       - 2fd45662e3d0ec0077ea2fa66b6378f0:0x6000039
+      - 54390bda109aab7fc006b8b4ead5b6c2
```
Comment (mr-tz) [inline collection/browser/gather-chrome-based-browser-login-information.yml:13]: can you include the offset here and in the other rule, please?

**Review by mr-tz — CHANGES_REQUESTED** [review]
Thanks! I've added comments to make the linter happy.

**Comment by mr-tz** [comment]
from the linter:
```
  FAIL: rule string quotes incorrect: add double quotes to "/\\+(Edge|Chrome|Chromium|Brave\-Browser|YandexBrowser|Kometa|Orbitum|Dragon|Torch|Amigo)\\+User Data\\+Default(\\+Network)?\\+(Cookies|Login Data)/i"
```

File: collection/browser/gather-chrome-based-browser-login-information.yml [hunk 2]
```diff
@@ -4,18 +4,21 @@ rule:
...
     scope: function
     att&ck:
       - Credential Access::Credentials from Password Stores::Credentials from Web Browsers [T1555.003]
     examples:
       - 2fd45662e3d0ec0077ea2fa66b6378f0:0x6000039
+      - 54390bda109aab7fc006b8b4ead5b6c2:0x1006E8D3
   features:
     - and:
       - or:
-        - string: /\\(Edge|Chrome|Chromium|Brave\-Browser|YandexBrowser|Kometa|Orbitum|Dragon|Torch|Amigo)\\User Data\\Default\\Login Data/
```
Comment (mr-tz) [inline collection/browser/gather-chrome-based-browser-login-information.yml:15]: I think we need `regex` here instead of `substring`?

**Comment by Still34** [comment]
Ah right, oops, accidentally used substring instead of string.

**Comment by mr-tz** [comment]
Now, we get this from the linter:
`  FAIL: doesn't match on referenced example: Fix the rule logic or provide a different example`

**Comment by Still34** [comment]
Sorry for the stale PR - it looks like the linter passes the test locally for me though.

```
 capa-rules  py -3 "E:\repos\capa\scripts\lint.py" --thorough  --samples E:\repos\capa\tests\data -t "gather chrome based browser login information" .
INFO:lint:successfully loaded 799 rules
INFO:lint:collecting potentially referenced samples
linting rule: gather chrome based browser log...: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  2.96rule/s]
INFO:lint:no lints failed, nice!
```

**Comment by mr-tz** [comment]
It doesn't match for me. The function doesn't contain the SELECT strings. Can you double check what's off?

**Comment by Still34** [comment]
After having a headache trying to figure out what went wrong for 30 minutes, finally figured out what was causing it to not match - it was due to the default scope the previous rule used, switching to `file` fixed it. It should pass the linter now *hopefully*.

**Review by mr-tz — APPROVED** [review]
looks good, thank you!

**Pull request merged**